base/shell/explorer/startup.cpp

```diff
@@ -259,17 +263,83 @@ static BOOL ProcessRunKeys(HKEY hkRoot, LPCWSTR szKeyName, BOOL bDelete,
     return res == ERROR_SUCCESS ? TRUE : FALSE;
 }
 
+static BOOL
+AutoStartupApplications(INT nCSIDL_Folder)
+{
+    WCHAR szPath[MAX_PATH] = { 0 };
+    HRESULT hResult;
+    HANDLE hFind;
+    WIN32_FIND_DATAW FoundData;
+    size_t cchPathLen;
+
+    TRACE("(%d)\n", nCSIDL_Folder);
+
+    // Get the special folder path
+    hResult = SHGetFolderPathW(NULL, nCSIDL_Folder, NULL, SHGFP_TYPE_CURRENT, szPath);
+    cchPathLen = wcslen(szPath);
+    if (!SUCCEEDED(hResult) || cchPathLen == 0)
+    {
+        WARN("SHGetFolderPath() failed with error %lu\n", GetLastError());
+        return FALSE;
+    }
+
+    // Build a path with wildcard
+    StringCbCatW(szPath, sizeof(szPath), L"\\*");
 
-int
-ProcessStartupItems(VOID)
+    // Start enumeration of files
+    hFind = FindFirstFileW(szPath, &FoundData);
+    if (hFind == INVALID_HANDLE_VALUE)
+    {
+        WARN("FindFirstFile(%s) failed with error %lu\n", debugstr_w(szPath), GetLastError());
+        return FALSE;
+    }
+
+    // Enumerate the files
+    do
+    {
+        // Ignore "." and ".."
+        if (wcscmp(FoundData.cFileName, L".") == 0 ||
+            wcscmp(FoundData.cFileName, L"..") == 0)
+        {
+            continue;
+        }
+
+        // Don't run hidden files
+        if (FoundData.dwFileAttributes & FILE_ATTRIBUTE_HIDDEN)
+            continue;
+
+        // Build the path
+        szPath[cchPathLen + 1] = UNICODE_NULL;
+        StringCbCatW(szPath, sizeof(szPath), FoundData.cFileName);
+
+        TRACE("Executing %s in directory %s\n", debugstr_w(FoundData.cFileName), debugstr_w(szPath));
+
+        DWORD dwType;
+        if (GetBinaryTypeW(szPath, &dwType) )
```

Comment:
Extra space between the closing parentheses is not needed.

Reply:
OK.